### DISCUSSION TITLE
can 128G RAM run two PreCommit1 task?

### DISCUSSION THREAD

**Discussion by olinkcloud** [filecoin-project/lotus#9395] [General Q&A]
Hello,
I saw that some tutorials said 128G memory can run two PreCommit1 task at the same time.
But when I try that, the miner is always killed due to OOM.

**Comment by stuberman** [comment]
Yes. 128GB ram will accommodate 2xPC1 concurrently.

**Reply by stuberman** [reply]
You need to look at what may be consuming more memory than needed. Are you running Ubuntu server (without desktop/GUI)?

**Reply by olinkcloud** [reply]
> You need to look at what may be consuming more memory than needed. Are you running Ubuntu server (without desktop/GUI)?

Yes, I’m running Ubuntu 22.04 server, I only run the lotus-node on this PC. It uses about 4GB RAM.

**Reply by stuberman** [reply]
I have not tried 22.04, I use 20.04.5 LTS (GNU/Linux 5.15.0-48-generic x86_64). I also use some swap  (200  GiB)
ulimit -n 1048576

### Here are my settings:
[env]
export MINER_API_INFO=xxxx
export MARKETS_API_INFO=xxxx
export LOTUS_ALLOW_TRUNCATED_LOG=1
export LOTUS_WORKER_PATH=/path/...
export BELLMAN_CUDA_NVCC_ARGS="--fatbin --gpu-architecture=sm_70 --generate-code=arch=compute_70,code=sm_70"
export NEPTUNE_CUDA_NVCC_ARGS="--fatbin --gpu-architecture=sm_70 --generate-code=arch=compute_70,code=sm_70"
export TMPDIR=/tmpfs/tmpdir
export FIL_PROOFS_PARAMETER_CACHE=/path
export FIL_PROOFS_PARENT_CACHE=/path
export FIL_PROOFS_USE_GPU_COLUMN_BUILDER=1
export FIL_PROOFS_USE_GPU_TREE_BUILDER=1
export FIL_PROOFS_SDR_PARENTS_CACHE_SIZE=1073741824
export RUST_BACKTRACE=full
export RUST_LOG=info
export RUSTFLAGS="-C target-cpu=native -g"
export FFI_BUILD_FROM_SOURCE=1
export RUST_GPU_TOOLS_CUSTOM_GPU="NVIDIA GeForce RTX 3090:10496"
export FIL_PROOFS_USE_MULTICORE_SDR=1
export LOTUS_PANIC_JOURNAL_LOOKBACK=1
export LOTUS_PANIC_REPORT_PATH=/tmpfs
export PATH=/usr/local/cuda-11.7/bin${PATH:+:${PATH}}
export LD_LIBRARY_PATH=/usr/local/cuda-11.7/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
export CUDA_HOME=/usr/local/cuda
export FFI_USE_CUDA=1
export PC1_32G_MAX_CONCURRENT=2
export AP_32G_MAX_PARALLELISM=0

**Reply by olinkcloud** [reply]
do you run the lotus-miner under non-root user?

**Reply by stuberman** [reply]
Yes, I run the miner as a non root user

**Comment by olinkcloud** [comment]
BTW, I enabled the multi core SDR.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
the answer is no. you need swap to run 2 PC1 in parallel with 128GB ram
